src/main/java/com/linecorp/armeria/server/http/HttpServerHandler.java

```diff
@@ -216,6 +222,19 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
             handledLastRequest = true;
         }
 
+        // check if CORS preflight must be returned, or if
+        // we need to forbid access because origin could not be validated
+        if (config.corsConfig() != null && config.corsConfig().isCorsSupportEnabled()) {
```

Comment:
If we ensure `corsConfig` is not null by using disabled `CorsConfig` instead of null, we do not need the null check anymore.